src/pykiso/lib/auxiliaries/can_auxiliary.py

```diff
@@ -0,0 +1,209 @@
+
+from pykiso.connector import CChannel
+from pykiso.lib.auxiliaries.communication_auxiliary import CommunicationAuxiliary
+from pykiso.interfaces.dt_auxiliary import (DTAuxiliaryInterface, open_connector, close_connector)
+from pykiso import Message
+from pykiso.can_message import CanMessage
+from pykiso.can_parser import CanMessageParser
+from pykiso.exceptions import AuxiliaryNotStarted
+
+from contextlib import ContextDecorator
+
+log = logging.getLogger(__name__)
+
+class CanAuxiliary(DTAuxiliaryInterface):
+    """Auxiliary is used for reading and writing can messages defined in dbc file """
+
+    def __init__(self, com: CChannel, dbc_file: str, **kwargs):
+        """Constructor.
+
+        :param com: CChannel that supports raw communication over CAN
+        """
+        self.tx_task_on = False
+        super().__init__(
+            is_proxy_capable=True, tx_task_on=True, rx_task_on=True, **kwargs
+        )
+        
+        self.recv_can_messages = {}
+        self.channel = com
+        path_to_dbf_file = dbc_file
+        self.parser = CanMessageParser(path_to_dbf_file)
+        self.can_messages = {}
+
+    @open_connector
+    def _create_auxiliary_instance(self) -> bool:
+        """Open the connector communication.
+
+        :return: True if the channel is correctly opened otherwise False
+        """
+        log.internal_info("Auxiliary instance created")
+        return True
+
+    @close_connector
+    def _delete_auxiliary_instance(self) -> bool:
+        """Close the connector communication.
+
+        :return: always True
+        """
+        log.internal_info("Auxiliary instance deleted")
+        return True
+
+    def _receive_message(self, timeout_in_s: float = 0.1) -> None:
+        """Trigger connector reception method and put received messages
+        in the queue.
+
+        :param timeout_in_s: maximum time in second to wait for a response
+        """
+        try:
+            rcv_data = self.channel.cc_receive(timeout=timeout_in_s)
+            if rcv_data.get("msg") is not None:
+                log.internal_debug(f"received message '{rcv_data}' from {self.channel}")
+                message_name = self.parser.dbc.get_message_by_frame_id(rcv_data["remote_id"]).name
+                message_signals = self.parser.decode(rcv_data["msg"], rcv_data["remote_id"])
+                message_timestamp = float(rcv_data.get("timestamp", 0))
+                old_can_message = self.can_messages.get(message_name, None)
+                if old_can_message is None:
+                    can_msg = CanMessage(message_name, message_signals, message_timestamp)
+                    self.can_messages[message_name] = can_msg
+                else:
+                    self.can_messages[message_name].name = message_name
+                    self.can_messages[message_name].signals = message_signals
+                    self.can_messages[message_name].time_stamp = message_timestamp
+        except (Exception, KeyError):
```

Comment:
Simply catching the `Exception` will already catch the `KeyError`, if you need a different error message you might want to handle them separately:

```python
except KeyError:
    log.exception("Some message")
except Exception:
    log.exception("My very generic message")

Reply:
Ok, i will separate them